### PR TITLE
Remove asserEqualIgnoreType from test_complex

### PR DESCRIPTION
**Pull request by VitalyFedyunin**
This is example of executing on #38095 tasks

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38195 Replace instance of  assertEqualIgnoreType
* #38152 Tests: Remove exact_dtype from test classes
* **#38124 Remove asserEqualIgnoreType from test_complex**
* #38123 Fix tests for non MKL enviroments
* #38103 assertEqual now requires matching dtypes
* #38102 Replacing assertEqual with assertEqualIgnoreType wherever types missmatch
* #34154 Introduce assertEqualIgnoreType to prevent type checking of assertEqual

Differential Revision: [D21484939](https://our.internmc.facebook.com/intern/diff/D21484939)

### CODE REVIEW

**Review by anjali411 — APPROVED** [review]

**Comment by VitalyFedyunin** [comment]
Looks like this tests are still failing, which makes me suspicious of implementation. @anjali411 what `copy_imag` supposed to do (we are missing docs here)?

File: test/test_complex.py [hunk 1]
```diff
@@ -30,10 +30,8 @@ def test_copy_real_imag_methods(self):
         real = torch.randn(4)
         imag = torch.randn(4)
         complex_tensor = real + 1j * imag
-        # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
-        self.assertEqualIgnoreType(complex_tensor.copy_real(), real)
-        # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
-        self.assertEqualIgnoreType(complex_tensor.copy_imag(), imag)
+        self.assertEqual(complex_tensor.copy_real(), real + 1j * 0)
+        self.assertEqual(complex_tensor.copy_imag(), 0 + 1j * imag)
```
Comment (anjali411) [inline test/test_complex.py:34]: `self.assertEqual(complex_tensor.copy_imag(), imag + 0j)`

`.copy_real()` and `.copy_imag()` return a new complex tensor with imaginary component equal to zero and real component equal to the real and imaginary values of the complex_tensor. These methods will ultimately be removed when storage becomes untyped and `real` and `imag` would then be made tensor properties which would perform a view.

Comment (VitalyFedyunin) [reply test/test_complex.py:34]: In this case (since tests are failing) I assume we have something wrong with this functions. I can take a look later.

Comment (anjali411) [reply test/test_complex.py:34]: you need to compare `complex_tensor.copy_imag()` with `imag + 0j` not `0 + 1j * imag`

Comment (anjali411) [reply test/test_complex.py:34]: alternately you could also do:
```
self.assertEqual(complex_tensor.copy_real().to(torch.float), real)
self.assertEqual(complex_tensor.copy_imag().to(torch.float, imag)
```

**Pull request closed without merging**